pkg/apis/machineconfiguration.openshift.io/v1/types.go

```diff
@@ -432,7 +432,7 @@ type ContainerRuntimeConfigSpec struct {
 // ContainerRuntimeConfiguration defines the tuneables of the container runtime
 type ContainerRuntimeConfiguration struct {
 	// pidsLimit specifies the maximum number of processes allowed in a container
-	PidsLimit int64 `json:"pidsLimit,omitempty"`
+	PidsLimit *int64 `json:"pidsLimit,omitempty"`
```

Comment:
Shouldn’t the CRD be updated to indicate the value is optional? (I don’t understand what are source files and what is generated, and I don’t know all that much about CRDs, this just looks curious to me.)

If `0` is a special case, shouldn’t it be documented? (BTW CRI-O does _not_ document 0 as a special case (but then again, runtime-spec/specs-go does document it), so it is curious that it should be supported here if it is undocumented in CRI-O.)

Reply:
It's been a long day, so pardon the simple question but why the change to *int64?

Reply:
> It's been a long day, so pardon the simple question but why the change to *int64?

Changed to *int64 to indicate whether pidsLimit is set in the containerruntimeconfig definition. Using int64 the decode of containerruntimeconfig definatoin can't differentiate if the configuration is set to 0 or there's no specific pidslimit configuration.

Reply:
> 
> If `0` is a special case, shouldn’t it be documented? (BTW CRI-O does _not_ document 0 as a special case (but then again, runtime-spec/specs-go does document it), so it is curious that it should be supported here if it is undocumented in CRI-O.)

I have' t found the documentation in CRIO but have checked with the crio team, the value will go down to runc, 0 is valid means no limit will apply.

Reply:
- I’m still curious about the CRD
- So, ideally _both_ CRI-O should document `0` (not part of this PR, sure), and this repo should document `0` (a new feature introduced by this PR, so should be documented by this PR), shouldn’t they?